pandas/core/arrays/arrow/array.py

```diff
@@ -866,11 +866,15 @@ def to_numpy(
 
         pa_type = self._data.type
         if (
-            is_object_dtype(dtype)
+            pa.types.is_time(pa_type)
```

Comment:
Could you use `pa.types.is_temporal` for this `if` check?

Reply:
We could use `is_temporal` but it would slow down the pyarrow date types. This is a slower path which is only really needed (ATM) for types with units and/or timezones (time, timestamp, duration). It would be nice to remove this block entirely, but I think that requires a bit more work elsewhere for consistency.

Reply:
Ah okay. Could you note that we don't want to use `is_temporal` here because it would slow down date types?